tests/gateway/test_client.py

```diff
@@ -0,0 +1,197 @@
+import pytest
+from requests.exceptions import HTTPError
+from unittest import mock
+
+from mlflow.gateway.envs import MLFLOW_GATEWAY_URI  # TODO: change to environment_variables import
+from mlflow.exceptions import MlflowException, InvalidUrlException
+import mlflow.gateway.utils
+from mlflow.gateway import set_gateway_uri, MlflowGatewayClient
+from mlflow.gateway.config import Route
+from mlflow.utils.databricks_utils import MlflowHostCreds
+from tests.gateway.tools import Gateway, store_conf
+
+
+@pytest.fixture
+def basic_config_dict():
+    return {
+        "routes": [
+            {
+                "name": "completions-gpt4",
+                "type": "llm/v1/completions",
+                "model": {
+                    "name": "gpt-4",
+                    "provider": "openai",
+                    "config": {
+                        "openai_api_key": "mykey",
+                        "openai_api_base": "https://api.openai.com/v1",
+                        "openai_api_version": "2023-05-10",
+                        "openai_api_type": "openai/v1/chat/completions",
+                    },
+                },
+            },
+            {
+                "name": "chat-gpt4",
+                "type": "llm/v1/chat",
+                "model": {
+                    "name": "gpt-4",
+                    "provider": "openai",
+                    "config": {"openai_api_key": "MY_API_KEY"},
+                },
+            },
+            {
+                "name": "claude-chat",
+                "type": "llm/v1/chat",
+                "model": {
+                    "name": "claude-v1",
+                    "provider": "anthropic",
+                    "config": {
+                        "anthropic_api_key": "api_key",
+                    },
+                },
+            },
+        ]
+    }
+
+
+@pytest.fixture
+def gateway(basic_config_dict, tmp_path):
+    conf = tmp_path / "config.yaml"
+    store_conf(conf, basic_config_dict)
+    with Gateway(conf) as g:
+        yield g
+    mlflow.gateway.utils._gateway_uri = None
```

Comment:
I'd create a separate autouse fixture.

Reply:
yep I missed that earlier comment that you made. It's fixed locally.